coriolis/osmorphing/windows.py

```diff
@@ -68,6 +69,39 @@
 REQUIRED_DETECTED_WINDOWS_OS_FIELDS = [
     "version_number", "edition_id", "installation_type", "product_name"]
 
+INTERFACES_PATH_FORMAT = (
+    "HKLM:\\%s\\ControlSet001\\Services\\Tcpip\\Parameters\\Interfaces")
+
+STATIC_IP_SCRIPT_TEMPLATE = """
+$NICS_INFO = ConvertFrom-Json "%(nics_info)s"
+$IPS_INFO = ConvertFrom-Json "%(ips_info)s"
+
+$i = 0
+foreach ($nic in $NICS_INFO) {
+    $mac_address = $nic.mac_address.replace(':', '-').toUpper()
+    $interface = Get-NetAdapter | Where MacAddress -eq $mac_address
+    if (-Not $interface) {
+        $interface = @(Get-NetAdapter | Where Status -eq Up)[$i]
+    }
+    $i++
+
+    Remove-NetIPAddress -InterfaceIndex $interface.ifIndex -Confirm:$false
+    foreach ($ip in $nic.ip_addresses) {
+        $ip_info = $IPS_INFO | Where ip_address -eq $ip
+        if ($ip_info.default_gateway) {
+            Get-NetRoute | Where NextHop -eq $ip_info.default_gateway | Remove-NetRoute -Confirm:$false
+            New-NetIPAddress -InterfaceIndex $interface.ifIndex -IPAddress $ip_info.ip_address -PrefixLength $ip_info.prefix_length -DefaultGateway $ip_info.default_gateway
+        } else {
+            New-NetIPAddress -InterfaceIndex $interface.ifIndex -IPAddress $ip_info.ip_address -PrefixLength $ip_info.prefix_length
+        }
+
+        if ($ip_info.dns_addresses) {
+            Set-DnsClientServerAddress -InterfaceIndex $interface.ifIndex -ServerAddresses $ip_info.dns_addresses
+        }
+    }
+}
```

Comment:
How/where will errors in the script be handled/logged?
If it won't be in a logical place (e.g. in the Cloudbase-init logs), please add a separate try-catch within this script and write the error to a file or something.

Reply:
The errors are logged in `Cloudbase-Init/Logs` dir.